File: astexpr/astexpr_java/src/com/iso/astexpr/VariableNode.java
package com.iso.astexpr;

public class VariableNode extends ASTNode
{
    private String name;

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public VariableNode(String name)
    {
        super(ASTNodeType.Variable);
        setName(name);
    }
}

File: astexpr/astexpr_cpp/astexpr/variablenode.cpp
#include "variablenode.h"


VariableNode::VariableNode(string name) :
	ASTNode(ASTNodeType::Variable),
	name(name)
{

}

VariableNode::~VariableNode()
{
	cout << "VariableNode destructor" << endl;
}

string VariableNode::GetName()
{
	return this->name;
}

